### PR TITLE
Add type hints, code refactor and improvements

### PR DESCRIPTION
**Pull request by javdc**
- Fix playback from URL found in YouTube Music
- Add type hints for better IDE suggestions
- Also use get_channel method in tts command
- Change VoiceChannel class name to CurrentVoiceChannel to avoid confusion with discord.py's VoiceChannel
- Add missing await in a message in ytmusic command
- Change OpenAI client variable name to avoid warnings in IDE
- Change max characters for Loquendo TTS to 600
- Move some methods from utils to their respective classes
- Add missing requests library to requirements.txt
- Fix periods and ellipsis in all messages
- Some more improvements to remove warnings and improve code readability

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 discord.py
 discord.py[voice]
+requests
```
Comment (Wikijito7) [inline requirements.txt:3]: is this needed?

File: youtube.py [hunk 2]
```diff
@@ -1,52 +1,59 @@
...
-def yt_search_and_extract_yt_dlp_info(search_query):
+def yt_search_and_extract_yt_dlp_info(search_query: str) -> Any:
     try:
         ydl_opts = {
-            'format': yt_dlp_formats,
-            'extract_flat': True, # Don't try to obtain information from nested content (very slow in long playlists)
-            'noplaylist': True, # Ensure playlists are discarded
+            'format': YT_DLP_FORMATS,
+            'extract_flat': True,  # Don't try to obtain information from nested content (very slow in long playlists)
+            'noplaylist': True,  # Ensure playlists are discarded
```
Comment (Wikijito7) [inline youtube.py:29]: extra white space

File: youtube.py [hunk 3]
```diff
@@ -1,52 +1,59 @@
...
     try:
         ydl_opts = {
-            'format': yt_dlp_formats,
-            'extract_flat': True, # Don't try to obtain information from nested content (very slow in long playlists)
-            'noplaylist': True, # We only want one video when searching
-            'playlist_items': '1', # Only get first item in lists
+            'format': YT_DLP_FORMATS,
+            'extract_flat': True,  # Don't try to obtain information from nested content (very slow in long playlists)
+            'noplaylist': True,  # We only want one video when searching
+            'playlist_items': '1',  # Only get first item in lists
```
Comment (Wikijito7) [inline youtube.py:45]: extra white space

**Review by Wikijito7 — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (javdc) [reply requirements.txt:3]: Yes

Comment (Wikijito7) [reply requirements.txt:3]: it was used without being added in requirements.

**Review by Wikijito7 — APPROVED** [review]
LGTM

File: tts.py [hunk 4]
```diff
@@ -39,36 +38,48 @@ def get_loquendo_tts(text):
...
     print("\r%d%%" % percent, end="")
 
 
-def generate_tts(text, speed, listener):
-    audio = None
-
-    if len(text) > 256:
-        audio = get_google_tts(text, speed)
+def generate_tts(text: str, listener: Callable[[str], Any]):
+    if len(text) > 600:
```
Comment (Wikijito7) [inline tts.py:60]: Restore back to original value

**Review by Wikijito7 — CHANGES_REQUESTED** [review]

File: tts.py (hunk 4 above)
Comment (javdc) [reply tts.py:60]: No

**Pull request merged**